### PR TITLE
mobile responsive

### LINKED ISSUE #42: Improve mobile support (feature)
It would be great to have improved support when accessing from a mobile, i.e. a simple responsive design.
I have the following suggestions:
- collapsible top menu instead of fixed sidebar
- larger font size
- disabling text overlay, I noticed a significant computational effort (but possibly that could also be the fast streaming)

### PR DESCRIPTION
**Pull request by ihatenumbers**
Resolves #42

Just makes it more readable in mobile view and added a settings button for the menu to appear. The button settings placement could probably placed in a better position and be somewhat bigger.

responsive1

Error: HTTP 400 happens when #50 isn't merged (using togetherai backend)

responsive2

### CODE REVIEW

**Comment by lmg-anon** [comment]
Nice idea, this looks very promising! However, I think these problems need to be solved before it can be merged:

1. Currently, it only works with the "Monospace Dark" theme.
2. The right margins are broken.
image
3. The bottom bar covers the text.
image
4. The text looks way too big.
image

**Comment by KaraKaraWitch** [comment]
I had an alterative implementation for the mobile responsive layout. Didn't push to any git repo but it should be simpler to implement.
```css
@media (max-width: 767.98px) {
	body {
		flex-direction: column;
	}
	#prompt-container {
		width:100%;
		max-width: 100%;
	}
	
	#sidebar {
		width:auto;
		max-height: calc(30vh - 8px);
	}
}
```

**Comment by lmg-anon** [comment]
@ihatenumbers Are you still planning to work on this PR?

**Comment by ihatenumbers** [comment]
> Nice idea, this looks very promising! However, I think these problems need to be solved before it can be merged:
> 
> 1. Currently, it only works with the "Monospace Dark" theme.
> 2. The right margins are broken.
>    image
> 3. The bottom bar covers the text.
>    image
> 4. The text looks way too big.
>    image

I think I ironed out all of those problems. Anyway, when I tried to edit `padding` in #prompt-overlay, it's being overridden by earlier `padding: 2em 3em`.

**Comment by lmg-anon** [comment]
1, 2 and 4 seem fine now. I guess you were talking about 3? One alternative I can think of is dynamically setting a margin to prompt-container according to the height of the sidebar.

**Comment by ihatenumbers** [comment]
> 1, 2, and 4 seem fine now. I assume you were referring to issue 3, right? One alternative I can think of is dynamically setting a margin-bottom to prompt-container according to the height of the sidebar.
> 
> Also, it might be better for the sidebar to start hidden on mobile devices because it might not be intuitive to leave the sidebar. And it looks like if you're on the desktop and hide the sidebar and expand the window, the sidebar remains hidden

Only added some margin bottom to prompt-container and other things, is this fine now? Also, I tried merging with head without edits and it broke

File: mikupad.html
Comment (lmg-anon) [inline mikupad.html:1017]: 28px seems to be barely anything here, that's why I suggested dynamically changing the margin-bottom to the height of the sidebar so it would be enough for all fonts/screen sizes:
image

File: mikupad.html
Comment (lmg-anon) [inline mikupad.html:3881]: Afaict this will add an event listener every react render, you could fix this by using `useEffect` but I think a better way to solve this problem would be to make a new css style like `.mobile-hide { display: none; }` and put it inside the media query, then theorically the elements with this rule would be hidden only on mobile.

**Review by lmg-anon — CHANGES_REQUESTED** [review]

**Comment by neCo2** [comment]
>Also, I tried pulling from head without edits and it broke

What's the error?

**Comment by ihatenumbers** [comment]
> 28px seems to be barely anything here, that's why I suggested dynamically changing the margin-bottom to the height of the sidebar so it would be enough for all fonts/sidebar sizes: image
> 
> Maybe you could use ResizeObserver for this ([example](https://github.com/ihatenumbers/mikupad/blob/abea415a134740fae8af12608a2cd69ed07cfa47/mikupad.html#L1693)), unless you have a better idea.

took me longer than I'd admit, it's first time interacting with react. Is this fine now?

**Comment by ihatenumbers** [comment]
> > Also, I tried pulling from head without edits and it broke
> 
> What's the error?

1. POST http://127.0.0.1:3000/all 404 (Not Found)
2. Uncaught (in promise) 404

**Comment by neCo2** [comment]
Sounds like there's an issue with the `loadAllFromDatabase` method of `ServerDBAdapter`, but that's unfortunately all I can say. The maintainer's gonna have to figure that one out.

**Comment by lmg-anon** [comment]
Looks like the API SelectBox disappeared:
image

>took me longer than I'd admit, it's first time interacting with react.
I hope the examples made it a bit easier to figure out :)

>unhelpful errors.
>POST http://127.0.0.1:3000/all 404 (Not Found)
>Uncaught (in promise) 404

Sounds like you're using a web server, but it isn't the mikupad server, or you're using an old version of the mikupad server. I guess I will add a version check and that would fix both cases.

File: mikupad.html
Comment (lmg-anon) [inline mikupad.html:3869]: This name is quite bad, you should call it `sidebarHeight`

File: mikupad.html
Comment (lmg-anon) [inline mikupad.html:3879]: you need to pass an empty array as the second parameter otherwise the useEffect hook [runs every render](https://stackoverflow.com/a/58579462)!
```suggestion
	}, []);
```

**Review by lmg-anon — CHANGES_REQUESTED** [review]

**Review by lmg-anon — CHANGES_REQUESTED** [review]

**Comment by lmg-anon** [comment]
This looks fine to merge now, thank you for your contribution!

**Pull request merged**
